storage/aptosdb/src/utils/truncation_helper.rs

```diff
@@ -262,27 +262,48 @@ fn truncate_transaction_accumulator(
 }
 
 fn truncate_ledger_db_single_batch(
-    ledger_db: &DB,
+    ledger_db: Arc<LedgerDb>,
     event_store: &EventStore,
     transaction_store: &TransactionStore,
     start_version: Version,
     end_version: Version,
 ) -> Result<()> {
-    let batch = SchemaBatch::new();
+    let batch = LedgerDbSchemaBatches::new();
 
-    delete_transaction_index_data(transaction_store, start_version, end_version, &batch)?;
-    delete_per_epoch_data(ledger_db, start_version, end_version, &batch)?;
+    delete_transaction_index_data(
+        transaction_store,
+        start_version,
+        end_version,
+        &batch.transaction_db_batches,
+    )?;
+    delete_per_epoch_data(
+        ledger_db.metadata_db(),
+        start_version,
+        end_version,
+        &batch.ledger_metadata_db_batches,
+    )?;
     delete_per_version_data(start_version, end_version, &batch)?;
 
-    event_store.prune_events(start_version, end_version, &batch)?;
+    event_store.prune_events(start_version, end_version, &batch.event_db_batches)?;
+
+    truncate_transaction_accumulator(
+        ledger_db.transaction_accumulator_db(),
+        start_version,
+        end_version,
+        &batch.transaction_accumulator_db_batches,
+    )?;
 
-    truncate_transaction_accumulator(ledger_db, start_version, end_version, &batch)?;
+    ledger_db.write_schemas(batch)?;
 
-    batch.put::<DbMetadataSchema>(
+    // Always update the overall commit progress in the end since the commit is no atomic across dbs
```

Comment:
Should you update the progress first for truncation?

Reply:
oh, got it.